### DISCUSSION TITLE
develop/how_to/benchmark_a_model/

### DISCUSSION THREAD

**Discussion by giscus** [roboflow/supervision#1706] [Docs]
# develop/how_to/benchmark_a_model/

A set of easy-to-use utilities that will come in handy in any computer vision project.

https://supervision.roboflow.com/develop/how_to/benchmark_a_model/

**Comment by David-rn** [comment]
Hi! Thanks for the great guide :fire:, it is quite easy to follow.

I haven't found anything on supervision, correct me if I'm wrong. Would it be possible to create something like a `MetricSession`? 

For example, I'm testing three different models, but I want to combine all their benchmarks in a final plot, or in a table like the one in Roboflow's Model Leaderboard.  The `MetricsSession` would be in charge of sorting the results from the different models that we input to ploting them afterwards.

**Reply by LinasKo** [reply]
Hi @David-rn,

Great to hear your thoughts!

No, indeed, there is no such concept at the moment, but you're right — aggregating and comparing metrics is the natural next step a user would take.

What does exist, is the `to_pandas` converters for each result. Stacking such dataframes should be fairly easy, but it'd be great if we had an automatic implementation.

I'll open an issue; we can discuss there.

**Reply by LinasKo** [reply]
Opened issue #1707.
